standard/conversions.md

```diff
@@ -887,13 +887,12 @@ Not every lambda expression can be converted to expression tree types. The conve
 
 An implicit conversion exists from a method group ([§11.2](expressions.md#112-expression-classifications)) to a compatible delegate type ([§19.4](delegates.md#194-delegate-compatibility)). If `D` is a delegate type, and `E` is an expression that is classified as a method group, then `D` is compatible with `E` if and only if `E` contains at least one method that is applicable in its normal form ([§11.6.4.2](expressions.md#11642-applicable-function-member)) to any argument list ([§11.6.2](expressions.md#1162-argument-lists)) having types and modifiers matching the parameter types and modifiers of `D`, as described in the following.
 
-The compile-time application of the conversion from a method group `E` to a delegate type `D` is described in the following. Note that the existence of an implicit conversion from `E` to `D` does not guarantee that the compile-time application of the conversion will succeed without error.
+The compile-time application of the conversion from a method group `E` to a delegate type `D` is described in the following. Note that the existence of an implicit conversion from `E` to `D` does not guarantee that the compile-time application of the conversion will succeed without error. Therefore, the rules for applicable function member (§11.6.4.2) require an identity conversion between the method return type and the delegate's return type.
```

Comment:
It may be worth highlighting that although the compatibility requirement means "An identity or implicit reference conversion exists from the return type of M to the return type of D", the requirement for an identity conversion is stronger than this. (It's easy to miss that it's *only* identity conversions, compared with "identity or implicit reference conversion".)

Reply:
Let's clarify that now:

```suggestion
The compile-time application of the conversion from a method group `E` to a delegate type `D` is described in the following. Note that the existence of an implicit conversion from `E` to `D` does not guarantee that the compile-time application of the conversion will succeed without error. Therefore, the rules for applicable function member (§11.6.4.2) require an identity or implicit reference conversion from the method return type to the delegate's return type.
```

Reply:
Hmm... possibly. I'd thought that *was* a deliberate difference, but it's not. For example, this code is fine:

```csharp
using System;

class Test
{
    static void Main()
    {
        Func<string, object> func = Method;
    }

    static string Method(string text)
    {
        return "";
    }
}
```

I think it would be useful to get an example of a case where the delegate conversion exists, but the implicit conversion from `E` to `D` fails at compile-time.

Reply:
I was trying a few things this morning, and I haven't found one for an implicit *reference* conversion. Other implicit conversions (like from `int` to `long`) do fail: 

```csharp
sing System;

class Test
{
    static void Main()
    {
        Func<long, object> func = Method; // Error: No overload for 'Method' matches delegate 'Func<long, object>'
    }

    static int Method(string text)
    {
        return 2;
    }
}
```

Reply:
Right - and it's fine for that to fail, because the delegate type isn't compatible under the rules of [19.4](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/delegates.md#194-delegate-compatibility), so the conversion doesn't exist according to the section in this PR (the paragraph at 888).

Reply:
Basically, it's possible that over time we've removed the existence of these "present but inapplicable" conversions.

Reply:
I think you're right. I did add a sentence in 11.6.4.2 to allow implicit reference conversions as well as identity conversions.

Reply:
Let's discuss this at the next meeting unless we have any brainwaves between now and then. I've a feeling I'm forgetting something, but I don't know what yet...